### PR TITLE
feat(admin_audit): Migrate to non-deprecated IEventDispatcher

### PR DESCRIPTION
**Pull request by nickvergessen**
One step closer for #38546 

- [x] Based on https://github.com/nextcloud/server/pull/39089

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/admin_audit/lib/AppInfo/Application.php [hunk 1]
```diff
@@ -109,8 +111,9 @@ private function registerHooks(IAuditLogger $logger,
 		$this->groupHooks($logger, $serverContainer->get(IGroupManager::class));
 		$this->authHooks($logger);
 
-		/** @var EventDispatcherInterface $eventDispatcher */
-		$eventDispatcher = $serverContainer->get(EventDispatcherInterface::class);
+
+		/** @var IEventDispatcher $eventDispatcher */
+		$eventDispatcher = $serverContainer->get(IEventDispatcher::class);
```
Comment (ChristophWurst) [inline apps/admin_audit/lib/AppInfo/Application.php:116]: move into `boot` and the earlier registration

**Review by ChristophWurst — COMMENTED** [review]
Let's do the full cleanup :)

File: apps/admin_audit/lib/AppInfo/Application.php (hunk 1 above)
Comment (nickvergessen) [reply apps/admin_audit/lib/AppInfo/Application.php:118]: PR welcome. I have 85 event dispatchers to go, before we can merge the dependency update and don't really want to get sidetracked too much with refactorings.

**Review by come-nc — APPROVED** [review]

File: apps/admin_audit/lib/AppInfo/Application.php (hunk 1 above)
Comment (ChristophWurst) [reply apps/admin_audit/lib/AppInfo/Application.php:118]: The private methods all register closures, not class listeners. This is a bigger change: https://github.com/nextcloud/server/issues/39118

**Review by ChristophWurst — APPROVED** [review]

**Comment by ChristophWurst** [comment]
Autoloader changes are probably caused by a new version of Composer

**Comment by nickvergessen** [comment]
Rebased after https://github.com/nextcloud/server/pull/39089

**Review by miaulalala — APPROVED** [review]

**Comment by nickvergessen** [comment]
Docs in https://github.com/nextcloud/documentation/pull/10858

**Pull request merged**